src/datasources/staking-api/kiln-api.service.ts

```diff
@@ -239,32 +254,18 @@ export class KilnApi implements IStakingApi {
    * @returns array of DeFi vault identifiers - `chainIdentifier_vault`, e.g. `eth_0x123`
    * @see https://docs.api.kiln.fi/reference/getdefinetworkstats
    */
-  private getDefiVaultIdentifier(args: {
-    // TODO: Can use this.chainId
-    chainId: string;
-    vault: `0x${string}`;
-  }): string {
-    // TODO: Type this in accordance with DefiVaultStatsChains, throwing if this.chainId isn't supported
-    const chainIdentifiers = {
-      '1': 'eth',
-      '42161': 'arb',
-      '56': 'bsc',
-      '137': 'matic',
-      '10': 'op',
-    };
-
-    // Note: cannot narrow without it being a separate type guard
-    const isDeFiSupportedChain = (
-      chainId: string,
-    ): chainId is keyof typeof chainIdentifiers => {
-      return chainId in chainIdentifiers;
-    };
+  private getDefiVaultIdentifier(vault: `0x${string}`): string {
+    const chain = Object.entries(KilnApi.DefiVaultStatsChains).find(
+      ([, chainId]) => {
+        return chainId === this.chainId;
+      },
+    );
 
-    if (isDeFiSupportedChain(args.chainId)) {
-      const chainIdentifier = chainIdentifiers[args.chainId];
-      return `${chainIdentifier}_${args.vault}`;
+    if (chain) {
+      const chainIdentifier = chain[0];
+      return `${chainIdentifier}_${vault}`;
     } else {
-      throw new Error(`${args.chainId} is not supported for DeFi by Kiln`);
+      throw new Error(`${this.chainId} is not supported for DeFi by Kiln`);
```

Comment:
Question: Do we show the Error message in reponse? If so, Do we need to include the details? For sure, We need the details for log purposes.

Reply:
Yes, [we create a "response error" from the error thrown](https://github.com/safe-global/safe-client-gateway/blob/kiln-api-instance-chainid/src/datasources/staking-api/kiln-api.service.ts#L186).

Reply:
It seems like we are not showing the error to the user:
[https://github.com/safe-global/safe-client-gateway/blob/d6cb3068ac228fe32962a9c41c6440ab9bc189a1/src/datasources/errors/http-error-factory.ts#L17](http-error-factory)

Reply:
That error is thrown and caught by the [`GlobalErrorFilter`](https://github.com/safe-global/safe-client-gateway/blob/kiln-api-instance-chainid/src/routes/common/filters/global-error.filter.ts#L0-L1) where it is logged.